pkg/endpoint/regeneration/owner.go

```diff
@@ -103,3 +106,149 @@ type EndpointUpdater interface {
 	// for a new observed flow with the given characteristics.
 	UpdateProxyStatistics(l7Protocol string, port uint16, ingress, request bool, verdict accesslog.FlowVerdict)
 }
+
+// PolicyRepository is the interface which needs to be implemented by the owner 
+// of a policy repository.
+type PolicyRepository interface {
+	GetRevision() uint64
+	GetPolicyCache() PolicyCache
+	RLock()
+	RUnlock()
+	GetSelectorCache() SelectorCache
+}
+
+type PolicyCache interface {
+	UpdatePolicy(*identity.Identity) error
+	Lookup(*identity.Identity) SelectorPolicy
+}
+
+type SelectorCache interface {
+	RemoveSelectors(CachedSelectorSlice, CachedSelectionUser)
+	RemoveSelector(CachedSelector, CachedSelectionUser)
+	ChangeUser(CachedSelector, CachedSelectionUser, CachedSelectionUser)
+	AddIdentitySelector(CachedSelectionUser, api.EndpointSelector) (CachedSelector, bool)
+	AddFQDNSelector(CachedSelectionUser, api.FQDNSelector) (CachedSelector, bool)
+}
+
+type PolicyL4Filter interface {
+	L7ParserType() string
```

Comment:
This should return `policy.L7ParserType` but then we will import policy into this package again. Any better ideais @joestringer?